source/common/logger/ur_logger.hpp

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+
+#ifndef UR_LOGGER_HPP
+#define UR_LOGGER_HPP 1
+
+#include <algorithm>
+#include <memory>
+
+#include "ur_logger_obj.hpp"
+#include "ur_util.hpp"
+
+namespace logger {
+extern Logger logger;
+
+template <typename... Args>
+inline void debug(const char *format, Args &&...args) {
+    logger.log(logger::Level::DEBUG, format, std::forward<Args>(args)...);
+}
+
+template <typename... Args>
+inline void info(const char *format, Args &&...args) {
+    logger.log(logger::Level::INFO, format, std::forward<Args>(args)...);
+}
+
+template <typename... Args>
+inline void warning(const char *format, Args &&...args) {
+    logger.log(logger::Level::WARN, format, std::forward<Args>(args)...);
+}
+
+template <typename... Args>
+inline void error(const char *format, Args &&...args) {
+    logger.log(logger::Level::ERR, format, std::forward<Args>(args)...);
+}
+
+inline void setLevel(logger::Level level) { logger.setLevel(level); }
+
+inline void setFlushLevel(logger::Level level) { logger.setFlushLevel(level); }
+
+inline auto create_logger(std::string logger_name) {
+    // env var example: UR_LOG_LOADER="level:info;flush:info;output:stdout"
+    std::transform(logger_name.begin(), logger_name.end(), logger_name.begin(),
+                   ::toupper);
+    std::stringstream env_var_name;
+    const auto default_level = logger::Level::QUIET;
+    const auto default_flush_level = logger::Level::ERR;
+    const std::string default_output = "stderr";
+
+    env_var_name << "UR_LOG_" << logger_name;
+    auto map = getenv_to_map(env_var_name.str().c_str());
+    if (!map.has_value()) {
+        return Logger(std::make_unique<logger::StderrSink>());
+    }
+
+    auto level = default_level;
+    auto kv = map->find("level");
+    if (kv != map->end()) {
+        auto value = kv->second[0];
+
+        try {
+            level = str_to_level(value);
+        } catch (const std::invalid_argument &e) {
+            std::cerr << "Error when creating a logger instance from environment variable" << e.what();
+            return Logger(std::make_unique<logger::StderrSink>());
+        }
+    }
+
+    auto flush_level = default_flush_level;
+    kv = map->find("flush");
+    if (kv != map->end()) {
+        auto value = kv->second.front();
+
+        try {
+            flush_level = str_to_level(value);
+        } catch (const std::invalid_argument &e) {
+            std::cerr << "Error when creating a logger instance from environment variable" << e.what();
+            return Logger(std::make_unique<logger::StderrSink>());
+        }
+    }
+
+    std::vector<std::string> values = {default_output};
+    std::unique_ptr<logger::Sink> sink;
+
+    kv = map->find("output");
+    if (kv != map->end()) {
+        values = kv->second;
+    }
+
+    try {
+        sink = values.size() == 2 ? sink_from_str(values[0], values[1])
+                                  : sink_from_str(values[0]);
+    } catch (const std::invalid_argument &e) {
```

Comment:
nit: instead of dotting the function around with try catch, you can do it once, for example by creating a new function. Especially since all the exceptions are exactly the same.